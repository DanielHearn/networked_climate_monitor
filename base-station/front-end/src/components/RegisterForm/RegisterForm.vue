<template>
  <form id="register" class="form" @submit="checkRegister" method="post">
    <div class="input-box">
      <p class="text">Email</p>
      <input
        type="email"
        class="input--text"
        name="email"
        v-model="email"
        placeholder="Email"
        tabindex="0"
      />
    </div>
    <div class="input-box">
      <p class="text">Password</p>
      <input
        type="password"
        class="input--text"
        name="password"
        v-model="password"
        placeholder="Password"
        tabindex="0"
      />
    </div>
    <div class="input-box">
      <p class="text">Confirm Password</p>
      <input
        type="password"
        class="input--text"
        name="confirmPassword"
        v-model="confirmPassword"
        placeholder="Confirm password"
        tabindex="0"
      />
    </div>
    <error-list v-if="errors.length" title="Errors:" v-bind:errors="errors" />
    <v-button
      :hierachyLevel="'primary'"
      :text="'Register'"
      type="submit"
      tabindex="0"
    />
  </form>
</template>

<script src="./RegisterForm.js"></script>

<style lang="scss" src="./RegisterForm.scss"></style>
