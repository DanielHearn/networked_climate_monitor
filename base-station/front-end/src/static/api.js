import { HTTP } from './http-common'

/**
 * Get account details from API
 * @param {string} accessToken - User's access token
 */
export const getAccount = accessToken => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.get('account', {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Update account details in API
 * @param {string} accessToken - User access token
 * @param {object} data - Updated user data
 */
export const patchAccount = (accessToken, data) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.patch('account', data, {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Attempt user login with API
 * @param {string} email - User's email
 * @param {string} password - User's password
 */
export const login = (email, password) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post('login', {
      email: email,
      password: password
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Attempt user registration with API
 * @param {string} email - User's email
 * @param {string} password - User's password
 */
export const register = (email, password) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post('account', {
      email: email,
      password: password
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Change the user's password
 * @param {string} resetToken - User's reset token
 * @param {string} password - User's password
 */
export const changePassword = (resetToken, password) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post('accounts/actions/change-password', {
      reset_token: resetToken,
      password: password
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Blacklist the access token from future logins
 * @param {string} accessToken - User's access token
 */
export const logoutAccessToken = accessToken => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post(
      'logout/access',
      {},
      {
        headers: { Authorization: 'Bearer ' + accessToken }
      }
    )
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Blacklist the refresh token from future logins
 * @param {string} accessToken - User's access token
 */
export const logoutRefreshToken = accessToken => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.post(
      'logout/refresh',
      {},
      {
        headers: { Authorization: 'Bearer ' + accessToken }
      }
    )
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Update sensor details in API
 * @param {string} accessToken - User's acess token
 * @param {number} sensorID - Sensor ID
 * @param {object} data - Updated sensor data
 */
export const patchSensor = (accessToken, sensorID, data) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.patch(`sensors/${sensorID}`, data, {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Get climate data for a specified sensor ID from the API
 * @param {string} accessToken - User's access token
 * @param {number} sensorID - Sensor ID
 * @param {number} quantity - Quantity of climate data points to be retrieved
 * @param {date} rangeStart - Start date of the time period
 * @param {date} rangeEnd - End date of the time period
 */
export const getClimateData = (
  accessToken,
  sensorID,
  quantity,
  rangeStart,
  rangeEnd
) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.get(
      `sensors/${sensorID}/climate-data?range_start=${rangeStart}&range_end=${rangeEnd}`,
      {
        headers: { Authorization: 'Bearer ' + accessToken }
      }
    )
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Delete the specified sensor using its sensor ID
 * @param {string} accessToken - User's access token
 * @param {number} sensorID - Sensor ID
 */
export const deleteSensor = (accessToken, sensorID) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.delete(`sensors/${sensorID}`, {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Delete the climate data from the specified sensor using its sensor ID
 * @param {string} accessToken - User's access token
 * @param {number} sensorID - Sensor ID
 */
export const deleteClimateData = (accessToken, sensorID) => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.delete(`sensors/${sensorID}/climate-data`, {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}

/**
 * Get all sensors from the API
 * @param {string} accessToken - User's access token
 */
export const getSensors = accessToken => {
  const promise = new Promise(function(resolve, reject) {
    HTTP.get('sensors', {
      headers: { Authorization: 'Bearer ' + accessToken }
    })
      .then(response => {
        resolve(response)
      })
      .catch(e => {
        reject(e)
      })
  })
  return promise
}
