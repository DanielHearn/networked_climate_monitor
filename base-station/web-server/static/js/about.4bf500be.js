(window["webpackJsonp"]=window["webpackJsonp"]||[]).push([["about"],{be4d:function(t,e,n){"use strict";n.r(e);var u=function(){var t=this,e=t.$createElement;t._self._c;return t._m(0)},r=[function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",[n("p",[t._v("404 - Not Found")])])}],a=n("2877"),l={},s=Object(a["a"])(l,u,r,!1,null,null,null);e["default"]=s.exports},f820:function(t,e,n){"use strict";n.r(e);var u=function(){var t=this,e=t.$createElement;t._self._c;return t._m(0)},r=[function(){var t=this,e=t.$createElement,n=t._self._c||e;return n("div",{staticClass:"about"},[n("h1",[t._v("This is an about page")])])}],a=n("2877"),l={},s=Object(a["a"])(l,u,r,!1,null,null,null);e["default"]=s.exports}}]);
//# sourceMappingURL=about.4bf500be.js.map