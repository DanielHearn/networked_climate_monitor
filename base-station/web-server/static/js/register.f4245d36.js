(window["webpackJsonp"]=window["webpackJsonp"]||[]).push([["register"],{"04de":function(t,e,s){"use strict";var r=s("aeaf"),a=s.n(r);a.a},"73cf":function(t,e,s){"use strict";s.r(e);var r=function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("main-panel",[s("template",{slot:"content"},[s("div",{staticClass:"content-column"},[s("h1",{staticClass:"title"},[t._v("Climate Monitor")]),s("h2",{staticClass:"heading"},[t._v("Account Registration")]),t.$store.state.user.email?[s("p",{staticClass:"text bold register-success"},[t._v(" The account has been successfully registered ")]),s("p",{staticClass:"text"},[t._v(" Complete the setup instructions to start using the climate monitor system. ")]),s("ol",{staticClass:"ordered-list"},[s("li",[s("p",{staticClass:"text"},[t._v(" Save the reset token below into a safe location such as in a text file or on paper. This token will be required to change or reset the account's password. ")]),s("span",{staticClass:"text bold"},[t._v("Reset token: "+t._s(t.$store.state.user.reset_token))])]),s("li",[s("p",{staticClass:"text"},[t._v(" Connect each sensor node to their battery. ")])]),s("li",[s("p",{staticClass:"text"},[t._v(" Plugin each sensor node's micro usb port to a USB port with a USB cable to charge the node's battery until the battery level on the dashboard is \"High\". ")])]),s("li",[s("p",{staticClass:"text"},[t._v(" Once the sensor nodes have been fully charged place the nodes in the areas that will be monitored. ")])]),s("li",[s("p",{staticClass:"text"},[t._v(" If the sensor nodes or climate data aren't appearing in the dashboard move the nodes closer to the base station so that they can communicate. ")])])]),s("router-link",{staticClass:"link",attrs:{to:"/dashboard"}},[t._v("Completed the setup? Click here to view the dashboard")])]:[s("p",{staticClass:"text"},[t._v(" Enter an email and password to register the account. ")]),s("register-form"),s("router-link",{staticClass:"link",attrs:{to:"/login"}},[t._v("Already registered? Click here to login")])]],2)])],2)},a=[],o=s("af63"),i=function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("form",{staticClass:"form",attrs:{id:"register",method:"post"},on:{submit:t.checkRegister}},[s("div",{staticClass:"input-box"},[s("p",{staticClass:"text"},[t._v("Email")]),s("input",{directives:[{name:"model",rawName:"v-model",value:t.email,expression:"email"}],staticClass:"input--text",attrs:{type:"email",name:"email",placeholder:"Email",tabindex:"0",autocomplete:"email"},domProps:{value:t.email},on:{input:function(e){e.target.composing||(t.email=e.target.value)}}})]),s("div",{staticClass:"input-box"},[s("p",{staticClass:"text"},[t._v("Password")]),s("input",{directives:[{name:"model",rawName:"v-model",value:t.password,expression:"password"}],staticClass:"input--text",attrs:{type:"password",name:"password",placeholder:"Password",tabindex:"0",autocomplete:"new-password"},domProps:{value:t.password},on:{input:function(e){e.target.composing||(t.password=e.target.value)}}})]),s("div",{staticClass:"input-box"},[s("p",{staticClass:"text"},[t._v("Confirm Password")]),s("input",{directives:[{name:"model",rawName:"v-model",value:t.confirmPassword,expression:"confirmPassword"}],staticClass:"input--text",attrs:{type:"password",name:"confirmPassword",placeholder:"Confirm password",tabindex:"0",autocomplete:"new-password"},domProps:{value:t.confirmPassword},on:{input:function(e){e.target.composing||(t.confirmPassword=e.target.value)}}})]),t.errors.length?s("error-list",{attrs:{title:"Errors:",errors:t.errors}}):t._e(),s("v-button",{attrs:{hierachyLevel:"primary",text:"Register",type:"submit",tabindex:"0"}})],1)},n=[],c=(s("c975"),s("ac1f"),s("5319"),s("93da")),l=s("ecf6"),d=s("6d06"),u=s("66ca"),p=s("c860"),h={name:"register-form",components:{ErrorList:c["a"],vButton:p["a"]},data:function(){return{email:"",password:"",confirmPassword:"",errors:[]}},methods:{checkRegister:function(t){t.preventDefault(),this.errors=[];var e=this.email,s=this.password,r=this.confirmPassword,a=!0;""===e?(this.errors.push("Enter a email."),a=!1):-1===e.indexOf("@")&&(this.errors.push("Enter a valid email."),a=!1),""===s?(this.errors.push("Enter a password."),a=!1):(s.length<8||s.length>40)&&(this.errors.push("Password must be between 8 and 40 characters long."),a=!1),s!==r&&(this.errors.push("Password and confirm password be identical."),a=!1),a&&this.register(e,s)},register:function(t,e){var s=this;this.$toasted.show("Sending register request"),Object(l["l"])(t,e).then((function(e){var r=e.data;if(r.status&&r.access_token&&r.refresh_token){var a=s.$store.state.user;a.logged_in=!0,a.access_token=r.access_token,a.refresh_token=r.refresh_token,a.email=t,Object(u["c"])(r.access_token),Object(u["d"])(r.refresh_token),Object(l["d"])(r.access_token).then((function(t){var e=t.data;if(e.status&&e.account){var r=s.$store.state.user;r.settings=JSON.parse(e.account.settings.replace(/'/g,'"')),s.$store.commit("setUser",r)}})).catch((function(t){console.warn(t)})),s.$store.commit("setUser",a),s.$toasted.show("Registered")}})).catch((function(t){t.response&&t.response.data&&t.response.data.errors?s.errors=Object(d["d"])(t.response.data.errors):s.errors.push(t)}))}}},m=h,f=(s("04de"),s("2877")),v=Object(f["a"])(m,i,n,!1,null,null,null),g=v.exports,w={name:"register",components:{MainPanel:o["a"],RegisterForm:g}},_=w,b=Object(f["a"])(_,r,a,!1,null,null,null);e["default"]=b.exports},"7ac5":function(t,e,s){"use strict";var r=s("e56d"),a=s.n(r);a.a},"93da":function(t,e,s){"use strict";var r=function(){var t=this,e=t.$createElement,s=t._self._c||e;return s("div",{staticClass:"error-list"},[t.title?s("p",{staticClass:"title"},[t._v(t._s(t.title))]):t._e(),s("ul",{staticClass:"errors"},t._l(t.errors,(function(e,r){return s("li",{key:r,staticClass:"error"},[s("p",[t._v(t._s(e))])])})),0)])},a=[],o={name:"error-list",props:{title:{type:String,default:"",required:!1},errors:{type:Array,default:[],required:!0}}},i=o,n=(s("7ac5"),s("2877")),c=Object(n["a"])(i,r,a,!1,null,null,null);e["a"]=c.exports},a640:function(t,e,s){"use strict";var r=s("d039");t.exports=function(t,e){var s=[][t];return!!s&&r((function(){s.call(null,e||function(){throw 1},1)}))}},aeaf:function(t,e,s){},c975:function(t,e,s){"use strict";var r=s("23e7"),a=s("4d64").indexOf,o=s("a640"),i=s("ae40"),n=[].indexOf,c=!!n&&1/[1].indexOf(1,-0)<0,l=o("indexOf"),d=i("indexOf",{ACCESSORS:!0,1:0});r({target:"Array",proto:!0,forced:c||!l||!d},{indexOf:function(t){return c?n.apply(this,arguments)||0:a(this,t,arguments.length>1?arguments[1]:void 0)}})},e56d:function(t,e,s){}}]);
//# sourceMappingURL=register.f4245d36.js.map