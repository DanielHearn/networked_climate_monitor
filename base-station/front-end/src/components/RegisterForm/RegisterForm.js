import ErrorList from './../ErrorList/ErrorList.vue'
import { register } from './../../static/api'
import { processErrors } from './../../static/helpers'
import vButton from './../vButton/vButton.vue'

// Displays a register form with email, password, and confirm password inputs
export default {
  name: 'register-form',
  components: {
    ErrorList,
    vButton
  },
  data: function() {
    return {
      email: '',
      password: '',
      confirmPassword: '',
      errors: []
    }
  },
  methods: {
    // Validate form inputs and attempt register request if all inputs are valid
    checkRegister: function(e) {
      e.preventDefault()
      this.errors = []

      const email = this.email
      const password = this.password
      const confirmPassword = this.confirmPassword
      const numOfRequiredInputs = 3
      const validInputs = []

      if (email === '') {
        this.errors.push('Enter an email.')
      } else if (email.indexOf('@') === -1) {
        this.errors.push('Enter a valid email.')
      } else {
        validInputs.push('email')
      }

      if (password === '') {
        this.errors.push('Enter a password.')
      } else if (password.length < 8 || password.length > 40) {
        this.errors.push('Password must be between 8 and 40 characters long.')
      } else {
        validInputs.push('password')
      }

      if (confirmPassword === '') {
        this.errors.push('Enter the password again to confirm.')
      } else if (password !== confirmPassword) {
        this.errors.push('Password and confirm password should be identical.')
      } else {
        validInputs.push('confirm_password')
      }

      if (numOfRequiredInputs === validInputs.length) {
        this.register(email, password)
      }
    },
    // Send register request to API and update user state if request is valid
    register: function(email, password) {
      this.$toasted.show('Sending register request')

      register(email, password)
        .then(response => {
          const data = response.data
          if (data.status && data.access_token && data.refresh_token) {
            data.logged_in = true
            data.email = email

            this.$store.dispatch('register', data, true)
            this.$toasted.show('Registered')
          }
        })
        .catch(e => {
          if (e.response && e.response.data && e.response.data.errors) {
            this.errors = processErrors(e.response.data.errors)
          } else {
            this.errors.push(e)
          }
        })
    }
  }
}
