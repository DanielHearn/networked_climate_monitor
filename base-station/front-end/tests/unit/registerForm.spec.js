import Vuex from 'vuex'
import { createLocalVue, shallowMount } from '@vue/test-utils'
import RegisterForm from './../../src/components/RegisterForm/RegisterForm.vue'
import { register } from './../../src/static/api.js'

const localVue = createLocalVue()
localVue.use(Vuex)
jest.mock('./../../src/static/api.js')

const actions = {
  register: jest.fn()
}

describe('RegisterForm.vue', () => {
  let wrapper
  let store

  beforeEach(() => {
    store = new Vuex.Store({ actions })
    wrapper = shallowMount(RegisterForm, {
      mocks: {
        $toasted: {
          show: jest.fn()
        },
        register: jest.fn()
      },
      stubs: ['router-link'],
      store,
      localVue
    })
  })

  afterEach(() => {
    wrapper.destroy()
    jest.resetAllMocks()
  })

  it('Success', async () => {
    const response = {
      data: {
        status: 'Success',
        access_token: '1234',
        refresh_token: '1234'
      }
    }
    register.mockResolvedValueOnce(response)

    const email = 'email@email.com'
    const password = 'password'
    const confirmPassword = password

    const emailInput = wrapper.find('input[name="email"]')
    emailInput.element.value = email
    emailInput.trigger('input')

    const passwordInput = wrapper.find('input[name="password"]')
    passwordInput.element.value = password
    passwordInput.trigger('input')

    const confirmPasswordInput = wrapper.find('input[name="confirmPassword"]')
    confirmPasswordInput.element.value = confirmPassword
    confirmPasswordInput.trigger('input')

    expect(wrapper.vm.email).toBe(email)
    expect(wrapper.vm.password).toBe(password)
    expect(wrapper.vm.confirmPassword).toBe(confirmPassword)

    wrapper.trigger('submit')

    await wrapper.vm.$nextTick()

    expect(actions.register).toHaveBeenCalledTimes(1)
    expect(wrapper.vm.errors).toEqual([])
  })

  it('Missing email', async () => {
    const password = 'password'
    const confirmPassword = password

    const passwordInput = wrapper.find('input[name="password"]')
    passwordInput.element.value = password
    passwordInput.trigger('input')

    const confirmPasswordInput = wrapper.find('input[name="confirmPassword"]')
    confirmPasswordInput.element.value = confirmPassword
    confirmPasswordInput.trigger('input')

    expect(wrapper.vm.email).toBe('')
    expect(wrapper.vm.password).toBe(password)
    expect(wrapper.vm.password).toBe(confirmPassword)

    wrapper.trigger('submit')

    await wrapper.vm.$nextTick()

    expect(actions.register).toHaveBeenCalledTimes(0)
    expect(wrapper.vm.errors).toEqual(['Enter an email.'])
  })

  it('Missing password', async () => {
    const email = 'email@email.com'
    const confirmPassword = '1234'

    const emailInput = wrapper.find('input[name="email"]')
    emailInput.element.value = email
    emailInput.trigger('input')

    const confirmPasswordInput = wrapper.find('input[name="confirmPassword"]')
    confirmPasswordInput.element.value = confirmPassword
    confirmPasswordInput.trigger('input')

    expect(wrapper.vm.email).toBe(email)
    expect(wrapper.vm.password).toBe('')
    expect(wrapper.vm.confirmPassword).toBe(confirmPassword)

    wrapper.trigger('submit')

    await wrapper.vm.$nextTick()

    expect(actions.register).toHaveBeenCalledTimes(0)
    expect(wrapper.vm.errors).toEqual([
      'Enter a password.',
      'Password and confirm password should be identical.'
    ])
  })

  it('Missing confirm password', async () => {
    const email = 'email@email.com'
    const password = 'password'

    const emailInput = wrapper.find('input[name="email"]')
    emailInput.element.value = email
    emailInput.trigger('input')

    const passwordInput = wrapper.find('input[name="password"]')
    passwordInput.element.value = password
    passwordInput.trigger('input')

    expect(wrapper.vm.email).toBe(email)
    expect(wrapper.vm.password).toBe(password)
    expect(wrapper.vm.confirmPassword).toBe('')

    wrapper.trigger('submit')

    await wrapper.vm.$nextTick()

    expect(actions.register).toHaveBeenCalledTimes(0)
    expect(wrapper.vm.errors).toEqual(['Enter the password again to confirm.'])
  })

  it('Missing email, password and confirm password', async () => {
    expect(wrapper.vm.email).toBe('')
    expect(wrapper.vm.password).toBe('')
    expect(wrapper.vm.confirmPassword).toBe('')

    wrapper.trigger('submit')

    await wrapper.vm.$nextTick()

    expect(actions.register).toHaveBeenCalledTimes(0)
    expect(actions.register).toHaveBeenCalledTimes(0)
    expect(wrapper.vm.errors).toEqual([
      'Enter an email.',
      'Enter a password.',
      'Enter the password again to confirm.'
    ])
  })

  it('Invalid email', async () => {
    const email = 'email.com'
    const password = 'password'
    const confirmPassword = password

    const emailInput = wrapper.find('input[name="email"]')
    emailInput.element.value = email
    emailInput.trigger('input')

    const passwordInput = wrapper.find('input[name="password"]')
    passwordInput.element.value = password
    passwordInput.trigger('input')

    const confirmPasswordInput = wrapper.find('input[name="confirmPassword"]')
    confirmPasswordInput.element.value = confirmPassword
    confirmPasswordInput.trigger('input')

    expect(wrapper.vm.email).toBe(email)
    expect(wrapper.vm.password).toBe(password)
    expect(wrapper.vm.confirmPassword).toBe(confirmPassword)

    wrapper.trigger('submit')

    await wrapper.vm.$nextTick()

    expect(actions.register).toHaveBeenCalledTimes(0)
    expect(wrapper.vm.errors).toEqual(['Enter a valid email.'])
  })

  it('Invalid password', async () => {
    const email = 'email@email.com'
    const password = 'pass'
    const confirmPassword = password

    const emailInput = wrapper.find('input[name="email"]')
    emailInput.element.value = email
    emailInput.trigger('input')

    const passwordInput = wrapper.find('input[name="password"]')
    passwordInput.element.value = password
    passwordInput.trigger('input')

    const confirmPasswordInput = wrapper.find('input[name="confirmPassword"]')
    confirmPasswordInput.element.value = confirmPassword
    confirmPasswordInput.trigger('input')

    expect(wrapper.vm.email).toBe(email)
    expect(wrapper.vm.password).toBe(password)
    expect(wrapper.vm.confirmPassword).toBe(confirmPassword)

    wrapper.trigger('submit')

    await wrapper.vm.$nextTick()

    expect(actions.register).toHaveBeenCalledTimes(0)
    expect(wrapper.vm.errors).toEqual([
      'Password must be between 8 and 40 characters long.'
    ])
  })

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
