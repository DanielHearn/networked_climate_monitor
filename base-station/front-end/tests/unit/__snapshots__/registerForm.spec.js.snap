// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm.vue has the expected html structure 1`] = `
<form
  class="form"
  id="register"
  method="post"
>
  <div
    class="input-box"
  >
    <p
      class="text"
    >
      Email
    </p>
     
    <input
      autocomplete="email"
      class="input--text"
      name="email"
      placeholder="Email"
      tabindex="0"
      type="email"
    />
  </div>
   
  <div
    class="input-box"
  >
    <p
      class="text"
    >
      Password
    </p>
     
    <input
      autocomplete="new-password"
      class="input--text"
      name="password"
      placeholder="Password"
      tabindex="0"
      type="password"
    />
  </div>
   
  <div
    class="input-box"
  >
    <p
      class="text"
    >
      Confirm Password
    </p>
     
    <input
      autocomplete="new-password"
      class="input--text"
      name="confirmPassword"
      placeholder="Confirm password"
      tabindex="0"
      type="password"
    />
  </div>
   
  <!---->
   
  <v-button-stub
    hierachylevel="primary"
    tabindex="0"
    text="Register"
    type="submit"
  />
</form>
`;

exports[`RegisterForm.vue has the expected html structure 1`] = `
<form
  class="form"
  id="register"
  method="post"
>
  <div
    class="input-box"
  >
    <p
      class="text"
    >
      Email
    </p>
     
    <input
      autocomplete="email"
      class="input--text"
      name="email"
      placeholder="Email"
      tabindex="0"
      type="email"
    />
  </div>
   
  <div
    class="input-box"
  >
    <p
      class="text"
    >
      Password
    </p>
     
    <input
      autocomplete="new-password"
      class="input--text"
      name="password"
      placeholder="Password"
      tabindex="0"
      type="password"
    />
  </div>
   
  <div
    class="input-box"
  >
    <p
      class="text"
    >
      Confirm Password
    </p>
     
    <input
      autocomplete="new-password"
      class="input--text"
      name="confirmPassword"
      placeholder="Confirm password"
      tabindex="0"
      type="password"
    />
  </div>
   
  <!---->
   
  <v-button-stub
    hierachylevel="primary"
    tabindex="0"
    text="Register"
    type="submit"
  />
</form>
`;
